dv/uvm/icache/doc/ibex_icache_dv_plan.md

```diff
@@ -1,99 +1,142 @@
 ---
-title: "IBEX_ICACHE DV Plan"
+title: "Ibex ICache DV Plan"
 ---
 
 ## Goals
 * **DV**
-  * Verify all IBEX_ICACHE IP features by running dynamic simulations with a SV/UVM based testbench
-  * Develop and run all tests based on the [testplan](#testplan) below towards closing code and functional coverage on the IP and all of its sub-modules
+  * Verify all Ibex ICache IP features by running dynamic simulations with a SV/UVM based testbench
+  * Develop and run all tests based on the [testplan](#testplan) below towards closing code and functional coverage.
 * **FPV**
-
-## Current status
-<!-- TODO for now -->
-* [Design & verification stage]({{< relref "doc/project/hw_dashboard" >}})
-  * [HW development stages]({{< relref "doc/project/hw_stages" >}})
-* [Simulation results](https://reports.opentitan.org/hw/ip/ibex_icache/dv/latest/results.html)
+  * [TBD]
 
 ## Design features
-For detailed information on IBEX_ICACHE design features, please see the [IBEX_ICACHE technical specification]({{< relref "doc/icache.rst" >}}).
+
+The ICache design is documented in the [Instruction Cache](https://ibex-core.readthedocs.io/en/latest/icache.html) section of the Ibex User Manual.
 
 ## Testbench architecture
-IBEX_ICACHE testbench has been constructed based on the [DV_LIB testbench architecture]({{< relref "vendor/lowrisc_ip/dv_lib/" >}}).
 
-### Block diagram
-![Block diagram](tb.svg)
+The testbench intentionally avoids knowing detailed information about the cache's performance characteristics (for example, cache size, line size or number of ways).
+This means that the testbench cannot compare the DUT with a reference model, nor can it model the exact requests that the DUT will make of instruction memory: the whole point of a cache is that it might avoid an instruction fetch.
+
+The cache has two main interfaces, which face the core and instruction bus respectively.
+We model this with two agents: the *core agent* and the *memory agent*.
+The core agent will emulate a core making instruction fetch requests from the cache and the memory agent will emulate the instruction bus.
```

Comment:
One thing that came up in a discussion earlier today - is there a way that we can create a base/generic memory agent that can then be extended for both ibex and the icache? The current ibex agent drives/monitors most of the signals relevant for the icache, so this could be extended in the icache to add the additional signals you plan to include in the cache memory interface.
This would definitely increase the reusability of the existing code, I'm just a little hesitant of having a new memory interface that would duplicate a lot of the already existing logic when we could stick this in a base driver/monitor to both benefit from it.
@sriyerg what do you think?

Reply:
For what it's worth, I've actually implemented most of this (because I wanted to make sure this would work). The interface is properly different from the memory interface that the core DV uses, because of the PMP line (this doesn't make it out to top-level). This has to be driven at a different time from the usual request/grant signals. I'm not sure that we'd gain anything by trying to combine these two different agents: while I haven't implemented analysis transactions in the monitor yet, I've done the slave stuff and the only code that I could directly copy from the slave memory interface was the 25 lines that drive the grant line.

Reply:
If you've already implemented it, then we can review the code and see if leveraging SV OOP is a better approach in terms of efficiency and reuse.

Reply:
Ah okay, if you already have some implementation of it, then we can review the code when you send out the PR, or we can spin off a side meeting if you want